### PR TITLE
mgis 2.1 (new formula)

### PR DESCRIPTION
**Pull request by maxence-wz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/mgis.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Mgis < Formula
+  desc "Provide tools to handle MFront generic interface behaviours"
+  homepage "https://thelfer.github.io/mgis/web/index.html"
+  license "GPL-1.0-or-later"
+  head "https://github.com/thelfer/MFrontGenericInterfaceSupport.git", using: :git, branch: "master"
```
Comment (chenrui333) [inline Formula/mgis.rb:5]:
```suggestion
  head "https://github.com/thelfer/MFrontGenericInterfaceSupport.git", branch: "master"
```

File: Formula/mgis.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  stable do
+    url "https://github.com/thelfer/MFrontGenericInterfaceSupport/archive/refs/tags/MFrontGenericInterfaceSupport-2.1.tar.gz"
+    sha256 "f5b556aab130da0c423f395fe4c35d6bf509dd8fc958242f2e37ea788464aea9"
```
Comment (chenrui333) [inline Formula/mgis.rb:8]: you can just move this before license line and remove `stable do`

File: Formula/mgis.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      "-Denable-enable-doxygen-doc=OFF",
+      "-Denable-c-bindings=ON",
+      "-Denable-fortran-bindings=ON",
+      "-Denable-python-bindings=ON",  # requires boost-python
+      "-Denable-fenics-bindings=OFF", # experimental and very limited
+      "-Denable-julia-bindings=OFF",  # requires CxxWrap library
+      "-Denable-enable-static=OFF",
+      "-Ddisable_python_library_linking=ON",
+      "-DCMAKE_INSTALL_RPATH=#{rpath}",
+    ]
```
Comment (chenrui333) [inline Formula/mgis.rb:30]:
```suggestion
    args = %W[
      -Denable-portable-build=ON
      -Denable-website=OFF
      -Denable-enable-doxygen-doc=OFF
      -Denable-c-bindings=ON
      -Denable-fortran-bindings=ON
      -Denable-python-bindings=ON  # requires boost-python
      -Denable-fenics-bindings=OFF # experimental and very limited
      -Denable-julia-bindings=OFF  # requires CxxWrap library
      -Denable-enable-static=OFF
      -Ddisable_python_library_linking=ON
      -DCMAKE_INSTALL_RPATH=#{rpath}
    ]
```

**Comment by maxence-wz** [comment]
@chenrui333 do you have any other requirements?

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/mgis.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+class Mgis < Formula
+  desc "Provide tools to handle MFront generic interface behaviours"
+  homepage "https://thelfer.github.io/mgis/web/index.html"
+  url "https://github.com/thelfer/MFrontGenericInterfaceSupport/archive/refs/tags/MFrontGenericInterfaceSupport-2.1.tar.gz"
+  sha256 "f5b556aab130da0c423f395fe4c35d6bf509dd8fc958242f2e37ea788464aea9"
+  license "GPL-1.0-or-later"
```
Comment (chenrui333) [inline Formula/mgis.rb:6]: is there any reference for this? looks like it should LGPL3??

```
/*!
 * \file   BehaviourData.cxx
 * \brief
 * \author Thomas Helfer
 * \date   02/08/2018
 * \copyright (C) Copyright Thomas Helfer 2018.
 * Use, modification and distribution are subject
 * to one of the following licences:
 * - GNU Lesser General Public License (LGPL), Version 3.0. (See accompanying
 *   file LGPL-3.0.txt)
 * - CECILL-C,  Version 1.0 (See accompanying files
 *   CeCILL-C_V1-en.txt and CeCILL-C_V1-fr.txt).
 */
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
sorry about the delay, lgtm now. (fixed the license to my understanding)

**Comment by chenrui333** [comment]
Also @maxence-wz thanks for your first PR to homebrew-core! 🎉

File: Formula/mgis.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  def install
+    args = %W[
+      -Denable-portable-build=ON
+      -Denable-website=OFF
+      -Denable-enable-doxygen-doc=OFF
+      -Denable-c-bindings=ON
+      -Denable-fortran-bindings=ON
+      -Denable-python-bindings=ON # requires boost-python
+      -Denable-fenics-bindings=OFF # experimental and very limited
+      -Denable-julia-bindings=OFF # requires CxxWrap library
```
Comment (ZhongRuoyu) [inline Formula/mgis.rb:28]: In Ruby's `%W` word arrays, `#` is not regarded as the beginning of a comment.

```console
$ ruby -e "%W[
arg1 # comment 1
arg2 # comment 2
].each { |s| puts s }
"
arg1
#
comment
1
arg2
#
comment
2
```

So what happened was basically:

```console
$ brew install -sv mgis
[output truncated]
==> cmake -S . -B build -Denable-portable-build=ON -Denable-website=OFF -Denable-enable-doxygen-doc=OFF -Denable-c-bindings=ON -Denable-fortran-bindings=ON -Denable-python-bindings=ON # requires boost-python -Denable-fenics-bindings=OFF # experimental and very limited -Denable-julia-bindings=OFF # requires CxxWrap library -Denable-enable-static=OFF -Ddisable_python_library_linking=ON -DCMAKE_INSTALL_RPATH=$ORIGIN/../lib -DCMAKE_INSTALL_PREFIX=/home/linuxbrew/.linuxbrew/Cellar/mgis/2.1 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF
CMake Warning:
  Ignoring extra path from command line:

   "library"

```

(I'm not sure how that worked, though. 🤷)

To add comments, you can use:

```ruby
args = [
  "arg1", # comment 1
  "arg2", # comment 2
]
```

Comment (ZhongRuoyu) [reply Formula/mgis.rb:28]: @maxence-wz Also, when you finish making changes, can you squash your commits into one and use the message `mgis 2.1 (new formula)`? Thanks!

File: Formula/mgis.rb (hunk 4 above)
Comment (maxence-wz) [reply Formula/mgis.rb]: yes, you're right

**Review by ZhongRuoyu — APPROVED** [review]
Thank you for your contribution, @maxence-wz!

**Pull request merged**